pkg/fleetautoscalers/fleetautoscalers.go

```
@@ -362,6 +367,97 @@ func applyCounterOrListPolicy(c *autoscalingv1.CounterPolicy, l *autoscalingv1.L
+			}
+		case autoscalingv1.WebhookPolicyType:
+			replicas, limited, err := applyWebhookPolicy(entry.Webhook, f)
+			// If no error was returned from the webhook policy, then return the values given
+			if err == nil {
+				return replicas, limited, nil
+			}
+		default:
+			// Every other policy type we just want to compute the desired fleet and return it
+			return computeDesiredFleetSize(entry.FleetAutoscalerPolicy, f, gameServerLister, nodeCounts)
+		}
+	}
+
+	// Fall off the chain
+	return replicas, limited, err
+}
+
+// isScheduleActive checks if a chain entry's is active and returns a boolean, true if active, false otherwise
+func isScheduleActive(s *autoscalingv1.SchedulePolicy, currentTime time.Time) bool {
+	// When retrieving the next available cronTime it returns the next minute e.g. if you want a policy to run every minute (all the time)
+	// If the current time is 1:00, the next start time returned will be 1:01, thus we use a cronDelta of 1 minute and 30 secpmds to
+	// adjust to the minute delta (since the next start time will always be ahead of the current time by 1 minute).
+	cronDelta := (time.Minute * -1) + (time.Second * -30)
+
+	// If a start time is present and the current time is before the start time, the schedule is inactive so return false
+	startTime := s.Between.Start.Time
+	if currentTime.Before(startTime) {
+		return false
+	}
+
+	// If an end time is present and the current time is after the end time, the schedule is inactive so return false
+	endTime := s.Between.End.Time
+	if !endTime.IsZero() && currentTime.After(endTime) {
+		return false
+	}
+
+	// If no startCron field is specified, then it's automatically true (duration is no longer relevant since we're always running)
+	if s.ActivePeriod.StartCron == "" {
+		return true
+	}
+
+	location, _ := time.LoadLocation(s.ActivePeriod.Timezone)
+	startCron, _ := cron.ParseStandard(s.ActivePeriod.StartCron)
+	nextStartTime := startCron.Next(currentTime.In(location)).Add(cronDelta)
```

All these fields are validated previously, right ?

---

Yep, both the location and startCron fields are validated prior to this.

---

you might document this - as a code reader I stumbled as to why it was okay to ignore the error.

a couple of other places we also just catch the error and log a bit error saying e.g. "Timezone already validated, unexpected error from time.LoadLocation: %v". Adding checks like that can serve as documentation instead of comments.